File: client/src/app/models/criteria/order-search-criteria.ts
import { OrderStatus } from '../enum/order-status';
import { AbstractSearchCriteria } from './abstract-search-criteria';

export class OrderSearchCriteria extends AbstractSearchCriteria {
  ids: number[] = [];
  deliveryArea: string;
  deliveryService: string;
  dispatchDate: Date;
  status: OrderStatus;
}

File: inventory-service/src/main/java/org/megamart/inventory/domain/criteria/OrderSearchCriteria.java
package org.megamart.inventory.domain.criteria;

import lombok.Getter;
import lombok.Setter;
import org.megamart.inventory.domain.enums.OrderStatus;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Order Search Criteria
 *
 * @author prasadm
 * @since 23 Feb 2021
 */

@Getter
@Setter
public class OrderSearchCriteria extends AbstractSearchCriteria
{
    private List<Long> ids = new ArrayList<>();
    private String deliveryArea;
    private String deliveryService;
    private Date dispatchDate;
    private OrderStatus status;
}
